processor/src/main/java/org/mapstruct/ap/internal/model/Field.java

```diff
@@ -105,4 +105,33 @@ public void setTypeRequiresImport(boolean typeRequiresImport) {
         this.typeRequiresImport = typeRequiresImport;
     }
 
+    @Override
+    public int hashCode() {
```

Comment:
Why only use the `variableName` in the `hashCode`? In `equals`, both the `variableName` and the `type` are used.